CorsixTH/Lua/world.lua

```diff
@@ -479,23 +479,28 @@ end
 --!return (boolean) Whether the disease is usable for new spawned patients.
 local function isDiseaseUsableForNewPatient(self, disease, hospital)
   if disease.only_emergency then return false end
-  if not disease.visuals_id then return true end
-
-  -- level files can delay visuals to a given month
-  -- and / or until a given number of patients have arrived
   local level_config = self.map.level_config
-  local hold_visual_months = level_config.gbv.HoldVisualMonths
-  local hold_visual_peep_count = level_config.gbv.HoldVisualPeepCount
+  if disease.visuals_id then
+    -- level files can delay visuals to a given month
+    -- and / or until a given number of patients have arrived
+    local hold_visual_months = level_config.gbv.HoldVisualMonths
+    local hold_visual_peep_count = level_config.gbv.HoldVisualPeepCount
+
+    -- if the month is greater than either of these values then visuals will not appear in the game
+    if (hold_visual_months and hold_visual_months > self.game_date:monthOfGame()) or
+        (hold_visual_peep_count and hold_visual_peep_count > hospital.num_visitors) then
+      return false
```

Comment:
```suggestion
      return error("disease has neither visuals_id or non_visuals_id")
```

Reply:
Actually, won't returning an error cause lua to think it returned true? Or will it still pick up it was false?

Reply:
L503 is what Cyco is referring to. As you elseif the non-visual there is a potential path that is unlikely to hit

Reply:
Yep got the line wrong. Though the question remains

Reply:
`error` never returns, it crashes the program with an error.
So no "return", just `error("crash message reason")`